app/src/main/java/me/phh/treble/app/HuaweiHeadsetUtils.kt

```diff
@@ -0,0 +1,34 @@
+package me.phh.treble.app
+
+
+import android.media.AudioManager
+import java.util.concurrent.locks.ReentrantLock
+import kotlin.concurrent.withLock
+
+object HuaweiHeadsetUtils {
+    private val lock = ReentrantLock()
+    private var mode: Int = AudioManager.MODE_INVALID
+    fun speaker(audioManager: AudioManager) {
+        lock.withLock {
```

Comment:
you could do synchronized(this), but that's a nit

Reply:
Kotlin has no synchronized(o). https://blog.egorand.me/concurrency-primitives-in-kotlin/

Reply:
Nevermind it does but still locks can provide guaranteed ordering and stuff.